### PR TITLE
Improved views, added pagination and search for products and orders

### PR DESCRIPTION
**Pull request by MateuszRoslak**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <%= csp_meta_tag %>
   <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
   <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/js/bootstrap.bundle.min.js" integrity="sha384-pprn3073KE6tl6bjs2QrFaJGz5/SUsLqktiwsUTF55Jfv3qYSDhgCecCxMW52nD2" crossorigin="anonymous"></script>
+  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.3/font/bootstrap-icons.css">
```
Comment (arturjaworski-htd) [inline app/views/layouts/application.html.erb:10]: I have added a bootstrap gem in the new PR. This line won't be necessary later.

File: app/views/orders/index.html.erb [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
-    <th scope="col">Order Number</th>
-    <th scope="col">Order Date</th>
-    <th scope="col">Quantity</th>
-    <th scope="col">Cost</th>
-    <th scope="col">Status</th>
+    <th scope="col"><%= sort_link(@q, :number, "Order Number",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :created_at, "Order Date",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :item_quantity, "Quantity",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :items_total, "Cost",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :status, "Status",{}, { class: "link-dark" }) %></th>
```
Comment (arturjaworski-htd) [inline app/views/orders/index.html.erb:35]: The name of `@q` variable should be changed.

File: app/views/orders/index.html.erb [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+  </div>
+  <div class="col">
+    <%= form_tag orders_path, :method => "get" do %>
+      <span>Orders per page:
+        <%= select_tag :per_page, options_for_select([5, 10, 20, 30], selected: params[:per_page] || 30), :class=>"select_box", :onchange => "this.form.submit()" %>
+    <% end %>
+    </span>
+
+  </div>
+</div>
```
Comment (arturjaworski-htd) [inline app/views/orders/index.html.erb:20]: It will be repetitive often, can't you make a method for it?

File: config/initializers/assets.rb [hunk 4]
```diff
@@ -10,3 +10,5 @@
 # application.js, application.scss, and all non-JS/CSS in the app/assets
 # folder are already added.
 # Rails.application.config.assets.precompile += %w( admin.js admin.css )
+
+Rails.application.config.assets.paths << Rails.root.join("node_modules/bootstrap-icons/font")
```
Comment (arturjaworski-htd) [inline config/initializers/assets.rb:14]: It is necessary?

File: app/controllers/orders_controller.rb [hunk 5]
```diff
@@ -4,7 +4,10 @@ class OrdersController < ApplicationController
   before_action :set_orders, only: [:index]
   before_action :set_order, only: [:show]
 
-  def index; end
+  def index
+    @q = current_user.orders.ransack(params[:q])
+    @orders = @q.result(distinct: true).paginate(page: params[:page], per_page: 10)
```
Comment (arturjaworski-htd) [inline app/controllers/orders_controller.rb:9]: The same, `@q` variable name.

File: app/controllers/cards_controller.rb [hunk 6]
```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
 
 class CardsController < ApplicationController
-  before_action :set_cards, only: [:show]
-  before_action :set_card, only: %i[destroy]
+  before_action :authenticate_user!
```
Comment (arturjaworski-htd) [inline app/controllers/cards_controller.rb:4]: `:authenticate_user!` is not necessary, we have it in `ApplicationController`

File: app/controllers/orders_controller.rb (hunk 5 above)
Comment (arturjaworski-htd) [reply app/controllers/orders_controller.rb:9]: Why do we have a constant of 10 and we don't take it from params? Same as in products. I prefer do helpers for sorting and pagination. I have sent you an example.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -18,7 +19,7 @@
   <%= render 'layouts/alerts' %>
 </header>
 <main>
-  <div class="container">
+  <div class="container pt-4 mt-5">
```
Comment (arturjaworski-htd) [inline app/views/layouts/application.html.erb:22]: Maybe set it in styles, in the container class.

File: app/views/orders/index.html.erb (hunk 2 above)
Comment (arturjaworski-htd) [reply app/views/orders/index.html.erb:35]: I recieve in the console: ``Passing two trailing hashes to `sort_link` is deprecated, merge the trailing hashes into a single one. ``

**Review by arturjaworski-htd — CHANGES_REQUESTED** [review]
Font size seems large:
image

**Comment by arturjaworski-htd** [comment]
Are these cards too big?
image

File: app/views/orders/index.html.erb [hunk 8]
```diff
@@ -1,11 +1,38 @@
...
-    <th scope="col">Order Number</th>
-    <th scope="col">Order Date</th>
-    <th scope="col">Quantity</th>
-    <th scope="col">Cost</th>
-    <th scope="col">Status</th>
+    <th scope="col"><%= sort_link(@query, :number, "Order Number", { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@query, :created_at, "Order Date", { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@query, :item_quantity, "Quantity", { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@query, :items_total, "Cost", { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@query, :status, "Status", { class: "link-dark" }) %></th>
```
Comment (arturjaworski-htd) [inline app/views/orders/index.html.erb:35]: What about `sort_link` warning?

File: app/views/products/index.html.erb [hunk 9]
```diff
@@ -1,5 +1,34 @@
...
+  <div class="col">
+    <span>Sort by:</span>
+    <%= sort_link(@query, :name, "Name", { class: "link-dark" }) %>
+    <%= sort_link(@query, :default_price, "Price", { class: "link-dark" }) %>
+  </div>
+  <div class="col">
+    <%= form_tag false, method: :get do %>
+    <span>Products per page:
+      <%= select_tag :per_page, options_for_select([5, 10, 20, 30], selected: params[:per_page] || 30), :class=>"select_box", :onchange => "this.form.submit()" %>
+    </span>
```
Comment (arturjaworski-htd) [inline app/views/products/index.html.erb:21]: We have to remember to refactor it in the future 👍

File: app/views/user/_session_manager.html.erb [hunk 10]
```diff
@@ -1,17 +1,37 @@
...
-    <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
-      <%= current_user.email %>
+    <a class="nav-link dropdown-toggle bi bi-person-circle text-light" title="Your account" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
     </a>
     <ul class="dropdown-menu dropdown-menu-lg-end" aria-labelledby="navbarDropdown">
-      <li><%= link_to "Orders", orders_path, class:"dropdown-item" %></li>
-      <li><%= link_to "Cards", cards_path, class:"dropdown-item" %></li>
+      <li>
+        <%= link_to orders_path, class:"dropdown-item" do %>
+          <h6><i class="bi bi-receipt"></i> Orders</h6>
```
Comment (arturjaworski-htd) [inline app/views/user/_session_manager.html.erb:16]: If we have pagination here, why we don't give page params here like in products ? `products_path(:per_page => 10)`

File: app/views/orders/index.html.erb (hunk 8 above)
Comment (MateuszRoslak) [reply app/views/orders/index.html.erb:35]: I can't get them on my browser. Thought that removing empty hash will fix the issue.

Comment (arturjaworski-htd) [reply app/views/orders/index.html.erb:35]: Ok, I don't see warnings now, my mistake 👍

File: app/views/layouts/_navbar.html.erb [hunk 11]
```diff
@@ -1,16 +1,15 @@
-<nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+<nav class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark h5">
```
Comment (arturjaworski-htd) [inline app/views/layouts/_navbar.html.erb:1]: `fixed-top` is necessary?  Flash messages appear under the navbar so we don't see them.

Comment (MateuszRoslak) [reply app/views/layouts/_navbar.html.erb:1]: No, I changed that because you mentioned it I think. Flash messages are not affected by margins?

**Pull request merged**